From f5303533fb64bf038dbd8203ac52d94819346239 Mon Sep 17 00:00:00 2001
From: Thomas Devoogdt <thomas.devoogdt@barco.com>
Date: Tue, 25 Jul 2023 10:00:36 +0200
Subject: [PATCH] CMakeLists.txt: disable cxx support for librdkafka #7741

Fluent-bit is c only, so no need to compile cxx.
This fixes also a compile error in buildroot.

Upstream: https://github.com/fluent/fluent-bit/pull/7765/commits
Signed-off-by: Thomas Devoogdt <thomas.devoogdt@barco.com>
---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index df6b283cf..10e6071f3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -967,6 +967,7 @@ if(FLB_BACKTRACE)
 endif()
 
 if(FLB_IN_KAFKA OR FLB_OUT_KAFKA)
+    FLB_OPTION(RDKAFKA_BUILD_CXX      Off)
     FLB_OPTION(RDKAFKA_BUILD_STATIC    On)
     FLB_OPTION(RDKAFKA_BUILD_EXAMPLES Off)
     FLB_OPTION(RDKAFKA_BUILD_TESTS    Off)
-- 
2.34.1

